### DISCUSSION TITLE
valid input

### DISCUSSION THREAD

**Discussion by BellBoy68** [Budibase/budibase#10122] [Help] [answered]
Hello.
I want to make sure that the user can only enter a phone number in the following format: +7-XXX-XXX-XX-XX
How to validate the form for correct data entry?

**Comment by melohagan** [comment, marked as answer]
Hey @BellBoy68 

It may be easiest to use the Input Mask custom component: https://github.com/melohagan/budibase-component-nikita-input-mask

**Reply by BellBoy68** [reply]
Is it possible to do without the plugin? It just doesn’t load for me and I also wanted to check passport data.

**Reply by melohagan** [reply]
![Screenshot 2023-03-27 at 15 51 43](https://user-images.githubusercontent.com/101575380/227977138-3b424afd-a9ee-419e-9ad2-6015551d9f3d.png)

![Screenshot 2023-03-27 at 15 52 13](https://user-images.githubusercontent.com/101575380/227977311-eea04e02-2901-4c4a-a528-a50524e493b3.png)

No issues adding it on my end!

**Reply by melohagan** [reply]
![Screenshot 2023-03-27 at 15 53 10](https://user-images.githubusercontent.com/101575380/227977623-1f1096b1-3acd-4548-a3c4-e3528889dd56.png)

**Reply by BellBoy68** [reply]
I get this error when adding.
![image](https://user-images.githubusercontent.com/127862266/228135289-14dd1a63-ea4f-41a5-968a-c4487c11257f.png)

**Reply by melohagan** [reply]
Hey @BellBoy68 I think you're using the wrong URL. 

You should select **GitHub** as the source, and then paste in the URL to the plugin repo as displayed here: https://github.com/melohagan/budibase-component-nikita-input-mask

**Reply by BellBoy68** [reply]
Still not working. But I solved my problem by validating through regex.
